libs/langchain/langchain/llms/databricks.py

```diff
@@ -52,14 +94,30 @@ def set_api_url(cls, values: Dict[str, Any]) -> Dict[str, Any]:
             values["api_url"] = api_url
         return values
 
-    def post(self, request: Any) -> Any:
-        # See https://docs.databricks.com/machine-learning/model-serving/score-model-serving-endpoints.html
-        wrapped_request = {"dataframe_records": [request]}
-        response = self.post_raw(wrapped_request)["predictions"]
-        # For a single-record query, the result is not a list.
-        if isinstance(response, list):
-            response = response[0]
-        return response
+    def post(
+        self, request: Any, transform_output_fn: Optional[Callable[..., str]] = None
+    ) -> Any:
+        if self.external_or_foundation:
+            resp = self.client.predict(endpoint=self.endpoint_name, inputs=request)
+            if transform_output_fn:
+                return transform_output_fn(resp)
+
+            if self.task == "llm/v1/chat":
+                return _transform_chat(resp)
+            elif self.task == "llm/v1/completions":
+                return _transform_completions(resp)
+
+            return resp
```

Comment:
@harupy Looks like this is still broken.

```
import mlflow
from langchain.chains import LLMChain
from langchain.prompts import PromptTemplate
from langchain.llms import Databricks

gateway = Databricks(
    endpoint_name="databricks-mpt-7b-instruct",
    # params={
    #     "temperature": 0.0,
    # },
)

llm_chain = LLMChain(
    llm=gateway,
    prompt=PromptTemplate(
        input_variables=["adjective"],
        template="Tell me a {adjective} joke",
    ),
)
result = llm_chain.run(adjective="funny")
```

```
HTTPError: 400 Client Error: Bad Request for url: https://oregon.staging.cloud.databricks.com/serving-endpoints/databricks-mpt-7b-instruct/invocations. Response text: Bad request: json: unknown field "dataframe_records"
```

I passed in a foundation model endpoint that should use the completions schema, but we're trying to send `dataframe_records`.

I also can't set params:

```
import mlflow
from langchain.chains import LLMChain
from langchain.prompts import PromptTemplate
from langchain.llms import Databricks

gateway = Databricks(
    endpoint_name="databricks-mpt-7b-instruct",
    params={
        "temperature": 0.0,
    },
)

llm_chain = LLMChain(
    llm=gateway,
    prompt=PromptTemplate(
        input_variables=["adjective"],
        template="Tell me a {adjective} joke",
    ),
)
result = llm_chain.run(adjective="funny")
```

```
ValidationError: 1 validation error for Databricks
params
  extra fields not permitted (type=value_error.extra)
```

Reply:
@dbczumar The first issue should be fixed by https://github.com/langchain-ai/langchain/pull/13699/commits/df54519db4d203cc195b07b426a5e6b2cff1dfdf.  `Databricks` doesn't have `params`. You need to use `model_kwargs` instead.

Reply:
It looks like The first issue is not fixed yet.

Reply:
@harupy `Databricks` should have `params` too for consistency :)